homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -95,48 +115,94 @@ class SynologyDSMFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
             host, port, username, password, use_ssl, dsm_version=api_version,
         )
 
-        if not await self.hass.async_add_executor_job(api.login):
-            errors[CONF_USERNAME] = "login"
-            return await self._show_setup_form(user_input, errors)
-
-        information: SynoDSMInformation = await self.hass.async_add_executor_job(
-            getattr, api, "information"
-        )
-        utilisation: SynoCoreUtilization = await self.hass.async_add_executor_job(
-            getattr, api, "utilisation"
-        )
-        storage: SynoStorage = await self.hass.async_add_executor_job(
-            getattr, api, "storage"
-        )
+        try:
+            serial = await self.hass.async_add_executor_job(
+                _login_and_fetch_syno_info, api
+            )
+        except InvalidAuth:
+            errors["base"] = "login"
+        except InvalidData:
+            errors["base"] = "missing_data"
 
-        if (
-            information.serial is None
-            or utilisation.cpu_user_load is None
-            or storage.disks_ids is None
-            or storage.volumes_ids is None
-        ):
-            errors["base"] = "unknown"
+        if "base" in errors:
```

Comment:
You can let `CONF_USERNAME` for the `InvalidAuth` error.

Then simply `if errors:` here.

So it will display the error on the right field

Reply:
nice, didn't realize that was built in :)